src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeSecurityHandles.cs

```diff
@@ -24,7 +22,7 @@ internal static SafeLsaMemoryHandle InvalidHandle
         {
             get
             {
-                return _invalidHandle;
+                return new SafeLsaMemoryHandle(IntPtr.Zero);
```

Comment:
The unwritten convention is for static `InvalidHandle` to return singleton and never allocate. This is changing to always allocate that makes it pointless.

Reply:
I see...but i don't understand other two helper in same file `SafeLsaPolicyHandle`  `SafeLsaReturnBufferHandle`

Reply:
It's unfortunate that this is changed to allocating, but looking at netfx, it seems many of these InvalidHandles are allocating, including this one in netfx (it must have been changed as part of the port).
https://referencesource.microsoft.com/#mscorlib/system/security/safesecurityhandles.cs,77

Reply:
I could add simple new here 
https://github.com/dotnet/corefx/blob/83ad4d3e5d14ba585dea88abb515f3bcca63ff94/src/System.Security.Principal.Windows/src/System/Security/Principal/NTAccount.cs#L240-L242
What do you think?

Reply:
I see that number of place deal with this problem by suppressing `Dispose` for the InvalidHandle. Maybe this one should do it too. The `SafeHandleCache` helper that they are using for it is performance de-optimization though that makes everything bigger and slower. That helper should better be deleted and replaced with simple static.

Alternative is to fix callers to not call Dispose on the InvalidHandle. A see a few places that are on this plan too.

Reply:
Do you proposo to re-add cached instance for all three safe handle and suppress dispose?If it's ok i'll go on!

Reply:
I removed Dispose call on the InvalidHandle of `SafeLsaMemoryHandle`, let me know if you want fix also for other handles.